### PR TITLE
Update zone.js to 0.5.1

### LINKED ISSUE #1862: Patch jasmine to execute the done() function in the root zone
The jasmine patch will exported when https://github.com/angular/zone.js/pull/105 lands and a new release of zone.js is available.

### PR DESCRIPTION
**Pull request by vicb**
TODO before merging:
- [x] update the commit comment (fix #1862)
- [x] update package.json with the zone.js release,
- [x] shrinkwrap

### CODE REVIEW

File: modules/angular2/test/test_lib/fake_async_spec.js [hunk 1]
```diff
@@ -40,17 +40,13 @@ export function main() {
...
-        }).toThrowError('fakeAsync() calls can not be nested');
-      });
-    }
+    it('should throw on nested calls', () => {
+      expect(() => {
+        fakeAsync(() => {
+          fakeAsync(() => null)();
+        })();
+      }).toThrowError('fakeAsync() calls can not be nested');
+    });
```
Comment (mhevery) [inline modules/angular2/test/test_lib/fake_async_spec.js]: I am not sure I understand how the `fakeAsync` should be used in a test. Could you provide an example?

Comment (vicb) [inline modules/angular2/test/test_lib/fake_async_spec.js]: Basically the same as you use `async()` in Angular1, this file is full of examples:

```
     it('should work with inject()', inject([Parser], fakeAsync((parser) => {
       expect(parser).toBeAnInstanceOf(Parser);
     })));

       it('should run queued zero duration timer on zero tick', fakeAsync(() => {
         var ran = false;
         PromiseWrapper.setTimeout(() => { ran = true }, 0);
 
         expect(ran).toEqual(false);
 
         tick();
         expect(ran).toEqual(true);
       }));
```

I guess your question comes from this particular test that is only here to test a corner case.

**Comment by vicb** [comment]
Blocked on zone.js release

**Comment by btford** [comment]
LGTM, assuming tests pass

**Comment by vicb** [comment]
@btford thanks, I'll wait for Travis before merging.

**Comment by vicb** [comment]
merged via 94272af45b5a2b8f28a7fabcf988ee849114642e
